lua/lspconfig/volar.lua

```diff
@@ -0,0 +1,103 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+local function get_typescript_server_path(root_dir)
+  local project_root = util.find_node_modules_ancestor(root_dir);
+  return project_root and (project_root .. '/node_modules/typescript/lib/tsserverlibrary.js') or '';
+end
+
+local server_name = 'volar'
+local bin_name = 'volar-server'
+
+-- https://github.com/johnsoncodehk/volar/blob/master/packages/shared/src/types.ts
+local volar_init_options = {
+  typescript = {
+    serverPath = get_typescript_server_path(vim.fn.getcwd()),
+  },
+  languageFeatures={
+    references = true,
+    definition = true,
+    typeDefinition = true,
+    callHierarchy = true,
+    hover = true,
+    rename = true,
+    renameFileRefactoring = true,
+    signatureHelp = true,
+    codeAction = true,
+    completion={
+      defaultTagNameCase='both',
+      defaultAttrNameCase='kebabCase',
+    },
+    schemaRequestService  = true ,
+    documentHighlight = true,
+    documentLink = true ,
+    codeLens = true,
+    semanticTokens = true,
+    diagnostics = true,
+  },
+  documentFeatures={
+    selectionRange = true,
+    foldingRange = true,
+    linkedEditingRange = true,
+    documentSymbol = true,
+    documentColor = true,
+    documentFormatting = {
+      defaultPrintWidth = 100,
+    },
+  }
+}
+
+-- https://github.com/johnsoncodehk/volar/blob/master/package.json#L56
+local volar_settings = {
+  volar= {
+    codeLens= {
+      references = true,
```

Comment:
@mjlbach Regarding your comment above about "is it necessary to set those explicitly". init_options need to be explicit, but volar_settings contains some things that basically set some settings to their default value. Here I'm setting references to true, but it's already true by default: https://github.com/johnsoncodehk/volar/blob/26b36fb01a83a68c94fcedd06144041876a7ae69/package.json#L78

I assume you'd like a minimal config that overrides defeaults only when it's necessary to make the LS work with nvim?

On the other hand, I'm worried people won't know about the settings = {} prop in nvim_lsp.volar.setup{}, since most LSs use init_options

Reply:
Would it be ok to do something like
```lua
init_options = volar_init_options,
-- https://github.com/johnsoncodehk/volar/blob/master/package.json
settings = {} 
```
?

Reply:
I would just use the package.json to autogenerate the documentation for these optiosn (see pyright or other servers, it's just one line with the url to the package.json). I find people get extremely confused when we duplicate default settings.

Reply:
https://github.com/neovim/nvim-lspconfig/pull/1223/commits/86431d9b5d09d1fd6d8be892d207b543fbe72265

Done. I also changed the link to package.json in a later commit, since the previous one didn't have a "contributes" property